src/engine/Server.cpp

```diff
@@ -510,26 +528,25 @@ CPP_template_2(typename RequestT, typename ResponseT)(
     // produced a `response`, send that now. Note that if multiple URL
     // parameters were processed, only the `response` from the last one is sent.
     if (response.has_value()) {
-      co_return co_await send(std::move(response.value()));
+      return send(std::move(response.value()));
     }
 
     // At this point, if there is a "?" in the query string, it means that there
     // are URL parameters which QLever does not know or did not process.
     if (request.target().find("?") != std::string::npos) {
-      throw std::runtime_error(
-          "Request with URL parameters, but none of them could be processed");
+      return send(createBadRequestResponse("Unknown query parameters",
+                                           std::move(request)));
```

Comment:
400 Bad Response is just more correct than a 500 ServerError. Also this was a place where we would get into the error handling without having a `PlannedQuery`. But still not possible for other reasons (see other thread).